asm-to-pil/src/vm_to_constrained.rs

```diff
@@ -278,9 +280,18 @@ impl<T: FieldElement> VMConverter<T> {
             input.pil.extend(self.pil);
         }
 
+        // This is hacky: in the absence of proof objects, we want to support both monolithic proofs and composite proofs.
+        // In the monolithic case, all degrees must be the same, so we align the degree of the rom to that of the vm.
+        // In the composite case, we set the minimum degree for the rom, which is the number of lines in the code.
+        let rom_degree = input
+            .degree
+            .clone()
+            .unwrap_or_else(|| Expression::from(self.code_lines.len() as u32));
```

Comment:
```suggestion
            .unwrap_or_else(|| Expression::from(self.code_lines.len().next_power_of_two() as u32));
```

Reply:
Or you can cherry-pick 2773609db21f947bd675c173977ea690d8ce0be7 to adjust the tests as well.

Reply:
done